pkg/network/usm/monitor_test.go

```diff
@@ -285,142 +344,209 @@ func (s *HTTPTestSuite) TestHTTPMonitorIntegrationWithNAT() {
-		EnableKeepAlive: true,
-	})
-	t.Cleanup(srvDoneFn)
+	for _, TCPTimestamp := range []struct {
+		name  string
+		value bool
+	}{
+		{name: "without TCP timestamp option", value: false},
+		{name: "with TCP timestamp option", value: true},
+	} {
+		t.Run(TCPTimestamp.name, func(t *testing.T) {
 
-	// Create a "raw" TCP socket that will serve as our HTTP client
-	// We do this in order to configure the socket option SO_LINGER
-	// so we can force a RST packet to be sent during termination
-	c, err := net.DialTimeout("tcp", serverAddr, 5*time.Second)
-	if err != nil {
-		t.Fatal(err)
-	}
+			monitor := newHTTPMonitor(t)
 
-	// Issue HTTP request
-	c.Write([]byte("GET /200/foobar HTTP/1.1\nHost: 127.0.0.1:8080\n\n"))
-	io.Copy(io.Discard, c)
+			serverAddr := "127.0.0.1:8080"
+			srvDoneFn := testutil.HTTPServer(t, serverAddr, testutil.Options{
+				EnableKeepAlive: true,
+			})
+			t.Cleanup(srvDoneFn)
 
-	// Configure SO_LINGER to 0 so that triggers an RST when the socket is terminated
-	require.NoError(t, c.(*net.TCPConn).SetLinger(0))
-	c.Close()
-	time.Sleep(100 * time.Millisecond)
+			// Create a "raw" TCP socket that will serve as our HTTP client
+			// We do this in order to configure the socket option SO_LINGER
+			// so we can force a RST packet to be sent during termination
+			c, err := net.DialTimeout("tcp", serverAddr, 5*time.Second)
+			if err != nil {
+				t.Fatal(err)
+			}
 
-	// Assert that the HTTP request was correctly handled despite its forceful termination
-	stats := monitor.GetHTTPStats()
-	url, err := url.Parse("http://127.0.0.1:8080/200/foobar")
-	require.NoError(t, err)
-	includesRequest(t, stats, &nethttp.Request{URL: url})
+			// Issue HTTP request
+			c.Write([]byte("GET /200/foobar HTTP/1.1\nHost: 127.0.0.1:8080\n\n"))
+			io.Copy(io.Discard, c)
+
+			// Configure SO_LINGER to 0 so that triggers an RST when the socket is terminated
+			require.NoError(t, c.(*net.TCPConn).SetLinger(0))
+			c.Close()
+			time.Sleep(100 * time.Millisecond)
+
+			// Assert that the HTTP request was correctly handled despite its forceful termination
+			stats := monitor.GetHTTPStats()
+			url, err := url.Parse("http://127.0.0.1:8080/200/foobar")
+			require.NoError(t, err)
+			includesRequest(t, stats, &nethttp.Request{URL: url})
+		})
+	}
 }
 
 func (s *HTTPTestSuite) TestKeepAliveWithIncompleteResponseRegression() {
 	t := s.T()
-	monitor := newHTTPMonitor(t)
 
-	const req = "GET /200/foobar HTTP/1.1\n"
-	const rsp = "HTTP/1.1 200 OK\n"
-	const serverAddr = "127.0.0.1:8080"
-
-	srvFn := func(c net.Conn) {
-		// emulates a half-transaction (beginning with a response)
-		n, err := c.Write([]byte(rsp))
-		require.NoError(t, err)
-		require.Equal(t, len(rsp), n)
+	for _, TCPTimestamp := range []struct {
+		name  string
+		value bool
+	}{
+		{name: "without TCP timestamp option", value: false},
+		{name: "with TCP timestamp option", value: true},
+	} {
+		t.Run(TCPTimestamp.name, func(t *testing.T) {
```

Comment:
@nplanel, I missed that in the previous reviews, but here you never use `TCPTimestamp.value`